routers/api/v1/repo/compare.go

```diff
@@ -0,0 +1,443 @@
+// Copyright 2019 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package repo
+
+import (
+	"net/http"
+	"path"
+	"strings"
+
+	"code.gitea.io/gitea/models"
+	"code.gitea.io/gitea/modules/base"
+	"code.gitea.io/gitea/modules/context"
+	"code.gitea.io/gitea/modules/git"
+	"code.gitea.io/gitea/modules/log"
+	"code.gitea.io/gitea/modules/setting"
+	api "code.gitea.io/gitea/modules/structs"
+	"code.gitea.io/gitea/services/gitdiff"
+)
+
+// GetCompare get a comparison of the two versions via shaes / branches / tags
+func GetCompare(ctx *context.APIContext) {
+	// swagger:operation GET /repos/{owner}/{repo}/git/compare/{<base sha | branch | tag>}...{[<head repo>:]<sha | branch | tag>} comparison of the two versions
```

Comment:
is there a reason to put it under /git/ ?

Reply:
Because its data source is the git diff command. so..., Because his data source is the git diff command, just my thoughts :>

Reply:
I'll have a closer look soon